lexers/c.lua

```diff
@@ -43,7 +45,9 @@ lex:add_rule('comment', lex:tag(lexer.COMMENT, line_comment + block_comment))
 
 -- Numbers.
 local integer = lexer.integer * lexer.word_match('u l ll ul ull lu llu', true)^-1
-local float = lexer.float * P('f')^-1
+local float = lexer.float * (P('f')+'l'+'F'+'L'+'df'+'dd'+'dl'+'DF'+'DD'+'DL'
```

Comment:
What do you think about using a `lexer.word_match()` like for integers above? It will allow for case-insensitive matching, which ought to be good enough. (It's a lexer, not a compiler.)

Reply:
Yeah, that'd work, I'll make the change